### PR TITLE
[ticket/15179] Update dependencies and fix compatibility with newer twig

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15179

### CODE REVIEW

File: phpBB/phpbb/path_helper.php [hunk 1]
```diff
@@ -108,9 +108,13 @@ public function update_web_root_path($path)
 			$path = $this->phpbb_root_path . substr($path, strlen($web_root_path));
 		}
 
-		if (strpos($path, $this->phpbb_root_path) === 0)
+		if (strpos($path, $this->phpbb_root_path) === 0 || $this->phpbb_root_path === './')
```
Comment (marc1706) [inline phpBB/phpbb/path_helper.php]: Not 100% happy with this but the current uses for update_web_root_path() are in append_sid() for relative paths and the assets for relative paths. The assets will be checked for allowed paths, append_sid() paths will not but are implied to be relative. I'll check what can be done there.

**Comment by Nicofuma** [comment]
why not, but needs exhaustive unit tests

**Comment by marc1706** [comment]
Update command:
```
php ../composer.phar update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing paragonie/random_compat (v1.4.1)
  - Installing paragonie/random_compat (v1.4.2)
    Loading from cache

  - Removing s9e/text-formatter (0.9.1)
  - Installing s9e/text-formatter (0.9.6)
    Downloading: 100%         

  - Removing symfony/filesystem (v2.8.14)
  - Installing symfony/filesystem (v2.8.21)
    Downloading: 100%         

  - Removing symfony/config (v2.8.14)
  - Installing symfony/config (v2.8.21)
    Downloading: 100%         

  - Removing symfony/debug (v2.8.14)
  - Installing symfony/debug (v2.8.21)
    Downloading: 100%         

  - Removing symfony/console (v2.8.14)
  - Installing symfony/console (v2.8.21)
    Downloading: 100%         

  - Removing symfony/finder (v2.8.14)
  - Installing symfony/finder (v2.8.21)
    Downloading: 100%         

  - Removing symfony/http-foundation (v2.8.14)
  - Installing symfony/http-foundation (v2.8.21)
    Downloading: 100%         

  - Removing symfony/event-dispatcher (v2.8.14)
  - Installing symfony/event-dispatcher (v2.8.21)
    Downloading: 100%         

  - Removing symfony/http-kernel (v2.8.14)
  - Installing symfony/http-kernel (v2.8.21)
    Downloading: 100%         

  - Removing symfony/dependency-injection (v2.8.14)
  - Installing symfony/dependency-injection (v2.8.21)
    Downloading: 100%         

  - Removing squizlabs/php_codesniffer (2.7.1)
  - Installing squizlabs/php_codesniffer (2.9.1)
    Loading from cache

  - Removing symfony/proxy-manager-bridge (v2.8.14)
  - Installing symfony/proxy-manager-bridge (v2.8.21)
    Downloading: 100%         

  - Removing symfony/routing (v2.8.14)
  - Installing symfony/routing (v2.8.21)
    Downloading: 100%         

  - Removing symfony/yaml (v2.8.14)
  - Installing symfony/yaml (v2.8.21)
    Downloading: 100%         

  - Removing sebastian/recursion-context (1.0.4)
  - Installing sebastian/recursion-context (1.0.5)
    Loading from cache

  - Removing sebastian/diff (1.4.1)
  - Installing sebastian/diff (1.4.3)
    Loading from cache

  - Removing sebastian/comparator (1.2.2)
  - Installing sebastian/comparator (1.2.4)
    Loading from cache

  - Removing phpunit/php-timer (1.0.8)
  - Installing phpunit/php-timer (1.0.9)
    Loading from cache

  - Removing phpunit/php-token-stream (1.4.9)
  - Installing phpunit/php-token-stream (1.4.11)
    Loading from cache

  - Removing phpdocumentor/reflection-docblock (2.0.4)
  - Installing phpdocumentor/reflection-docblock (2.0.5)
    Downloading: 100%         

  - Removing phpspec/prophecy (v1.6.2)
  - Installing phpspec/prophecy (v1.7.0)
    Loading from cache

  - Removing phpunit/phpunit (4.8.31)
  - Installing phpunit/phpunit (4.8.35)
    Loading from cache

  - Removing symfony/dom-crawler (v2.8.14)
  - Installing symfony/dom-crawler (v2.8.21)
    Downloading: 100%         

  - Removing symfony/browser-kit (v2.8.14)
  - Installing symfony/browser-kit (v2.8.21)
    Downloading: 100%         

  - Removing symfony/css-selector (v2.8.14)
  - Installing symfony/css-selector (v2.8.21)
    Downloading: 100%         

  - Removing symfony/process (v2.8.14)
  - Installing symfony/process (v2.8.21)
    Downloading: 100%         

  - Removing react/promise (v2.4.1)
  - Installing react/promise (v2.5.1)
    Loading from cache

Writing lock file
Generating autoload files
> echo 'You MUST manually modify the clean-vendor-dir target in build/build.xml when adding or upgrading dependencies.'
You MUST manually modify the clean-vendor-dir target in build/build.xml when adding or upgrading dependencies.

```

**Comment by marc1706** [comment]
Run after updating twig as well:
```
php ../composer.phar update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing twig/twig (v1.24.2)
  - Installing twig/twig (v1.33.2)
    Loading from cache

  - Removing symfony/twig-bridge (v2.8.11)
  - Installing symfony/twig-bridge (v2.8.21)
    Downloading: 100%         

Writing lock file
Generating autoload files
> echo 'You MUST manually modify the clean-vendor-dir target in build/build.xml when adding or upgrading dependencies.'
You MUST manually modify the clean-vendor-dir target in build/build.xml when adding or upgrading dependencies.
```

**Comment by marc1706** [comment]
Twig confirmed to be properly working now with extensions. Thx @rxu

**Comment by Nicofuma** [comment]
There is a nasty conflict with master, we need to run a composer update there

**Comment by rxu** [comment]
Twig 1.34.3 is out which is PHP 7.2 compatible (<a href="https://github.com/twigphp/Twig/blob/v1.34.3/CHANGELOG">Changelog<>).
It'd be nice to update to simplify firther "master" PHP 7.2 compatibility testing 😃

**Pull request merged**